### PR TITLE
Fixing 'vagrant up' command for m1 mac

### PR DESCRIPTION
**Pull request by bineesh-n**

### CODE REVIEW

File: lib/vagrant-parallels/action/sane_defaults.rb [hunk 1]
```diff
@@ -28,21 +28,23 @@ def call(env)
...
+            # longer_battery_life: 'on',
+            # shared_cloud: 'off',
+            # shared_profile: 'off',
+            # smart_mount: 'off',
+            # sh_app_guest_to_host: 'off',
+            # sh_app_host_to_guest: 'off',
+            # startup_view: 'headless',
+            # time_sync: 'on',
+            # disable_timezone_sync: 'on',
+            # shf_host_defined: 'off'
```
Comment (legal90) [inline lib/vagrant-parallels/action/sane_defaults.rb:47]: **question:** Are all of these settings not supported for "macvm" machines? 

In this case the better solution would be:
```ruby
def default_settings
  # Options defined below are not supported for `*.macvm` VMs
  return {} if is_macvm(@env)

  {
    tools_autoupdate: 'no',
    on_shutdown: 'close',
    on_window_close: 'keep-running',
    auto_share_camera: 'off',
    smart_guard: 'off',
    longer_battery_life: 'on',
    shared_cloud: 'off',
    shared_profile: 'off',
    smart_mount: 'off',
    sh_app_guest_to_host: 'off',
    sh_app_host_to_guest: 'off',
    startup_view: 'headless',
    time_sync: 'on',
    disable_timezone_sync: 'on',
    shf_host_defined: 'off'
  }
end
```

**Comment by bineesh-n** [comment]
Hi @legal90 Thanks for your comments. I've updated the code as per your suggestions now.

File: lib/vagrant-parallels/action/prepare_clone_snapshot.rb [hunk 2]
```diff
@@ -19,6 +20,12 @@ def call(env)
             return @app.call(env)
           end
 
+          if is_macvm(env)
+            #Ignore, since macvm doesn't support snapshot creation
+            @logger.info('Snapshot creation is not supported yet for macOS ARM, skip snapshot preparing')
```
[the remark this answers is not in the document]
Comment (polonius) [reply lib/vagrant-parallels/action/prepare_clone_snapshot.rb]: We support snapshots for VMs on ARM-based Macs.
Snapshots are not supported for macVms (.macvm) only.

**Review by legal90 — APPROVED** [review]

**Comment by legal90** [comment]
Thank you @bineesh-n !  I will merge it now. 
I'll do my best to verify it and release a new provider version later this week.

P.s. To Parallels folks: 
It's really nice to see more people engaged in this project now 👍 . Feel free to reach out to me if you want to add more maintainers to this repo. Otherwise, I can keep maintaining it on my spare time as I have been doing this for the last years .

**Comment by legal90** [comment]
Released in https://github.com/Parallels/vagrant-parallels/releases/tag/v2.2.6

**Pull request merged**